### PR TITLE
[Catalog] Moved Objective C code for catalog into a pod spec.

### PR DESCRIPTION
**Pull request by randallli**
This is so we can remove MDCCatalogTiles from the bridging header.
Internal tools do not support bridging headers.

### CODE REVIEW

File: catalog/MDCCatalogObjectiveC/MDCCatalogObjectiveC.podspec [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Pod::Spec.new do |s|
+  s.name         = "MDCCatalogObjectiveC"
+  s.version      = "38.0.1"
+  s.summary      = "A helper objective c classs for the MDC catalog."
```
Comment (jverkoey) [inline catalog/MDCCatalogObjectiveC/MDCCatalogObjectiveC.podspec]: "Helper Objective-C classes"

File: catalog/MDCCatalogObjectiveC/MDCCatalogObjectiveC.podspec [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Pod::Spec.new do |s|
+  s.name         = "MDCCatalogObjectiveC"
```
Comment (jverkoey) [inline catalog/MDCCatalogObjectiveC/MDCCatalogObjectiveC.podspec]: Can this simply be called "MaterialCatalog"?

**Review by jverkoey — CHANGES_REQUESTED** [review]

File: catalog/MDCCatalogObjectiveC/MDCCatalogObjectiveC.podspec (hunk 2 above)
Comment (randallli) [reply catalog/MDCCatalogObjectiveC/MDCCatalogObjectiveC.podspec]: Done.

File: catalog/MDCCatalogObjectiveC/MDCCatalogObjectiveC.podspec (hunk 1 above)
Comment (randallli) [reply catalog/MDCCatalogObjectiveC/MDCCatalogObjectiveC.podspec]: Done.

**Comment by randallli** [comment]
ptal

**Comment by randallli** [comment]
@jverkoey ptal. I applied your changes requested.

**Review by jverkoey — APPROVED** [review]

**Comment by jverkoey** [comment]
You may want to update the on-disk path to match the new podspec name as well.

**Pull request merged**